app/src/test/java/com/antosik/benchproject/app/common/state/UIStateTest.kt

```diff
@@ -0,0 +1,65 @@
+package com.antosik.benchproject.app.common.state
+
+import io.mockk.mockk
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+
+internal class UIStateTest {
+    companion object {
+        val data = mockk<Any>()
+
+        @JvmStatic
+        fun uiStatesParameters() = listOf(
+            Arguments.of(
+                Param(
+                    state = UIState.Loading,
+                    isLoading = true,
+                    isEmpty = false,
+                    isRefreshed = false,
+                    data = null
+                )
+            ),
+            Arguments.of(
+                Param(
+                    state = UIState.Empty,
+                    isLoading = false,
+                    isEmpty = true,
+                    isRefreshed = true,
+                    data = null
+                )
+            ),
+            Arguments.of(
+                Param(
+                    state = UIState.Success(data),
+                    isLoading = false,
+                    isEmpty = false,
+                    isRefreshed = true,
+                    data = data
+                )
+            ),
+        )
+    }
+
+    @ParameterizedTest
+    @MethodSource("uiStatesParameters")
```

Comment:
I think `DynamicTest` would be simpler that `MethodSource` here

Reply:
Done 😄